iocore/net/quic/QUICFrameRetransmitter.cc

```diff
@@ -128,3 +131,40 @@ QUICFrameRetransmitter::_create_stream_frame(QUICFrameInformationUPtr &info, uin
   ink_assert(frame != nullptr);
   return frame;
 }
+
+QUICFrameUPtr
+QUICFrameRetransmitter::_create_crypto_frame(QUICFrameInformationUPtr &info, uint16_t maximum_frame_size,
+                                             std::deque<QUICFrameInformationUPtr> &tmp_queue, QUICFrameId id,
+                                             QUICFrameGenerator *owner)
+{
+  CryptoFrameInfo *crypto_info = reinterpret_cast<CryptoFrameInfo *>(info->data);
+  // FIXME: has_offset and has_length should be configurable.
+  auto frame = QUICFrameFactory::create_crypto_frame(crypto_info->block, crypto_info->offset, id, owner);
+  if (frame->size() > maximum_frame_size) {
```

Comment:
I'm not familiar enough with CRYPTO frames, but I have a little concern on splitting a CRYPTO frame into multiple frames. Is it allowed, @masaori335 ?

Reply:
Basically, we can. TLS messages has its own length, so we don't need to care about the align on CRYPTO frame level.
But when ATS (client) retransmit first CRYPTO frame, it should not be splitted.

> [7. Cryptographic and Transport Handshake]
> The first CRYPTO frame from a client MUST be sent in a single packet.

Reply:
Thanks. The first packet use the minimum size (because we don't know MTU yet), so we should be able to retransmit the frame without splitting on following packets.